packages/firebase_remote_config/firebase_remote_config/lib/src/remote_config.dart

```diff
@@ -0,0 +1,159 @@
+      return _firebaseRemoteConfigInstances[app.name];
+    }
+
+    RemoteConfig newInstance = RemoteConfig._(app: app);
+    _firebaseRemoteConfigInstances[app.name] = newInstance;
+
+    return newInstance;
+  }
+
+  /// Returns the [DateTime] of the last successful fetch.
+  ///
+  /// If no successful fetch has been made a [DateTime] representing
+  /// the epoch (1970-01-01 UTC) is returned.
+  DateTime get lastFetchTime {
+    return _delegate.lastFetchTime;
+  }
+
+  /// Returns the status of the last fetch attempt.
+  RemoteConfigFetchStatus get lastFetchStatus {
+    return _delegate.lastFetchStatus;
+  }
+
+  /// Returns the [RemoteConfigSettings] of the current instance.
+  RemoteConfigSettings get settings {
+    return _delegate.settings;
+  }
+
+  /// Makes the last fetched config available to getters.
+  ///
+  /// Returns a [bool] that is true if the config parameters
+  /// were activated. Returns a [bool] that is false if the
+  /// config parameters were already activated.
+  Future<bool> activate() async {
+    bool configChanged = await _delegate.activate();
+    notifyListeners();
+    return configChanged;
+  }
+
+  /// Ensures the last activated config are available to getters.
+  Future<void> ensureInitialized() {
+    return _delegate.ensureInitialized();
+  }
+
+  /// Fetches and caches configuration from the Remote Config service.
+  Future<void> fetch() {
+    return _delegate.fetch();
+  }
+
+  /// Performs a fetch and activate operation, as a convenience.
+  ///
+  /// Returns [bool] in the same way that is done for [activate].
+  Future<bool> fetchAndActivate() async {
+    bool configChanged = await _delegate.fetchAndActivate();
+    notifyListeners();
+    return configChanged;
+  }
+
+  /// Returns a Map of all Remote Config parameters.
+  Map<String, RemoteConfigValue> getAll() {
+    return _delegate.getAll();
+  }
+
+  /// Gets the value for a given key as a bool.
+  bool getBool(String key) {
+    assert(key != null);
+    return _delegate.getBool(key);
+  }
+
+  /// Gets the value for a given key as an int.
+  int getInt(String key) {
+    assert(key != null);
+    return _delegate.getInt(key);
+  }
+
+  /// Gets the value for a given key as a double.
+  double getDouble(String key) {
+    assert(key != null);
+    return _delegate.getDouble(key);
+  }
+
+  /// Gets the value for a given key as a String.
+  String getString(String key) {
+    assert(key != null);
+    return _delegate.getString(key);
+  }
+
+  /// Gets the [RemoteConfigValue] for a given key.
+  RemoteConfigValue getValue(String key) {
+    assert(key != null);
+    return _delegate.getValue(key);
+  }
+
+  /// Sets the [RemoteConfigSettings] for the current instance.
+  Future<void> setConfigSettings(RemoteConfigSettings remoteConfigSettings) {
+    assert(remoteConfigSettings != null);
+    assert(remoteConfigSettings.fetchTimeout != null);
+    assert(remoteConfigSettings.minimumFetchInterval != null);
```

Comment:
Might be best to add another assert here to ensure the timeout value isn't less than 0.

Reply:
Yeah may be best check that both durations are not negative.